### ISSUE TITLE
Enable all remaining --incompatible flags once bazel catches up

### ISSUE THREAD

**Issue by Profpatsch** [tweag/rules_haskell#635]
@laurentlb FYI, most of the incompatible changes have been fixed in our codebase. 

I had to disable a couple of checks though:

- Both `--incompatible_new_actions_api` and `--incompatible_disable_deprecated_attr_params` are failing on the protobuf library.
- `--incompatible_expand_directories` is triggering a pretty weird behavior. When activated, it creates a cycle in the dependency graph.

```
$ bazel build //... --incompatible_expand_directories 
INFO: Invocation ID: 9c4b56cc-4bb2-450d-b79a-2d12f2478332
INFO: Analysed 217 targets (286 packages loaded, 4935 targets configured).
INFO: Found 217 targets...
ERROR: rules_haskell/tools/runfiles/BUILD:7:1: in haskell_library rule //tools/runfiles:runfiles: cycle in dependency graph:
    configured target: //tools/runfiles:bin
    file: _middlemen/tools_Srunfiles_Sbin-runfiles
    action from: ActionLookupData{actionLookupKey=//tools/runfiles:bin BuildConfigurationValue.Key[6d2637c3c06d6024aad2781cf0f375e3] false, actionIndex=14}
    file: tools/runfiles/libHStoolsZSrunfilesZSrunfiles-ghc8.6.3.so
    action from: ActionLookupData{actionLookupKey=//tools/runfiles:runfiles BuildConfigurationValue.Key[6d2637c3c06d6024aad2781cf0f375e3] false, actionIndex=11}
    file: tools/runfiles/_obj/runfiles.dynamic.manifest
    action from: ActionLookupData{actionLookupKey=//tools/runfiles:runfiles BuildConfigurationValue.Key[6d2637c3c06d6024aad2781cf0f375e3] false, actionIndex=8}
    file: tools/runfiles/_obj/runfiles
.-> action from: ActionLookupData{actionLookupKey=//tools/runfiles:runfiles BuildConfigurationValue.Key[6d2637c3c06d6024aad2781cf0f375e3] false, actionIndex=2}
|   file: tools/runfiles/extra_args_runfiles_HaskellBuildLibrary
|   action from: ActionLookupData{actionLookupKey=//tools/runfiles:runfiles BuildConfigurationValue.Key[6d2637c3c06d6024aad2781cf0f375e3] false, actionIndex=1}
|   file: tools/runfiles/toolsZSrunfilesZSrunfiles/_iface
`-- action from: ActionLookupData{actionLookupKey=//tools/runfiles:runfiles BuildConfigurationValue.Key[6d2637c3c06d6024aad2781cf0f375e3] false, actionIndex=2}
INFO: Elapsed time: 7.512s, Critical Path: 0.54s
INFO: 23 processes: 23 linux-sandbox.
FAILED: Build did NOT complete successfully
```

You can reproduce this error with rules haskell set to `316f858fc6aebf9140cfebf6befe97b3bd55e7c0`

_Originally posted by @NinjaTrappeur in https://github.com/tweag/rules_haskell/issues/489#issuecomment-456722164_
